app/pages/stacking/delegated-stacking/components/choose-membership-duration.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FC } from 'react';
+
+import { StackingStepBaseProps } from '../../utils/abstract-stacking-step';
+import {
+  StackingStep as Step,
+  StackingStepAction as Action,
+  StackingStepDescription as Description,
+} from '../../components/stacking-form-step';
+import { DurationSelectItem } from './duration-select-item';
+import { DurationSelect } from './duration-select';
+import { DurationCyclesForm } from './duration-cycles-form';
+import { useFormik } from 'formik';
+
+interface ChooseMembershipDurationStepProps extends StackingStepBaseProps {
+  description: string;
+  value?: string;
+  onEdit(): void;
+  onComplete(duration: number): void;
+}
+
+export const ChooseMembershipDurationStep: FC<ChooseMembershipDurationStepProps> = props => {
+  const { isComplete, description, state, step, title, value, onEdit, onComplete } = props;
+
+  const durationForm = useFormik({
+    initialValues: { duration: -1 },
+    onSubmit: ({ duration }) => onComplete(duration),
+  });
+
+  return (
+    <Step
+      title={title}
+      step={step}
+      state={state}
+      value={value}
+      isComplete={isComplete}
+      onEdit={onEdit}
+    >
+      <Description>{description}</Description>
+      <form onSubmit={durationForm.handleSubmit}>
+        <DurationSelect>
+          <DurationSelectItem
+            title="Indefinite"
+            durationType={-1}
+            duration={-1}
+            activeDuration={durationForm.values.duration}
+            index={0}
+            onChange={() => (durationForm.values.duration = -1)}
```

Comment:
There's a `.setValues` you should use rather than mutating the values object directly